### DISCUSSION TITLE
What is the condition for the if statement in this code?

### DISCUSSION THREAD

**Discussion by tzSharing** [vlang/v#18749] [General]
Who can tell me what the condition of this if statement is?
Is it `true == (v: = m ['abc'])`? or something else
I am confused.

```
fn main() {
	m := {
		'abc': 'def'
	}
	if v := m['abc'] {   //Confused me
		println('v=${v}')
	}
}

output：
v=def
```

**Comment by Delta456** [comment]
It is asserting if the value is there or not in the map.

**Reply by JalonSolov** [reply]
Think of it as "if exists" without an extra keyword.  If the key exists, the value will be stored in variable `v`.  Otherwise, `v` will be unset.

**Reply by tzSharing** [reply]
It seems to be syntactic sugar or fixed collocation. It's convenient to use but not very intuitive and easy to read. I'm still more used to using `if key in map {}`.

**Reply by JalonSolov** [reply]
It is "syntax sugar" for not having to use an `or` block to handle a possible error.

**Comment by ttytm** [comment]
It can be seen as rusts `if let` equivalent. Though I understand that it might be confusing at first. Thinking back of starting with rust it took some applications to get my head around the concept but then it felt like a luxury.

A simple example of an if let assignment for an optional value.
```rust
let number = Some(7);
if let Some(i) = number {
	println!("Matched {i}!");
}
```

instead of 
```rust
let number = Some(7);
if number.is_some() {
	println!("Matched {}!", number.unwrap());
}
```

V's way is easier to grasp I think.

It's concise flow control to make use of values that might be optional or errors or to make sure that values of map keys or array indices are existent.
